lib/socks.c

```diff
@@ -99,19 +99,12 @@ int Curl_blockread_all(struct Curl_easy *data,   /* transfer */
 }
 #endif
 
-#ifndef DEBUGBUILD
-#define sxstate(x,y) socksstate(x,y)
-#else
 #define sxstate(x,y) socksstate(x,y, __LINE__)
-#endif
-
 
 /* always use this function to change state, to make debugging easier */
 static void socksstate(struct Curl_easy *data,
-                       enum connect_t state
-#ifdef DEBUGBUILD
-                       , int lineno
-#endif
+                       enum connect_t state,
+                       int lineno
```

Comment:
Why do you think passing on the line number unconditionally is a good idea? Why is that needed to silence warnings?

Reply:
I do not consider it necessary to remove the alert.  I think this is the easier way to get rid of this warning.  Of course, you can also add an appropriate condition for the presence of an additional argument to the function. Which version will be better?

Reply:
I imagined something like this for the socks.c fix:
~~~diff
diff --git a/lib/socks.c b/lib/socks.c
index 91c4223a5..0bc0cfb76 100644
--- a/lib/socks.c
+++ b/lib/socks.c
@@ -97,28 +97,29 @@ int Curl_blockread_all(struct Curl_easy *data,   /* transfer */
   }
   return result;
 }
 #endif
 
-#ifndef DEBUGBUILD
-#define sxstate(x,y) socksstate(x,y)
-#else
+#if defined(DEBUGBUILD) && !defined(CURL_DISABLE_VERBOSE_STRINGS)
+#define DEBUG_AND_VERBOSE
 #define sxstate(x,y) socksstate(x,y, __LINE__)
+#else
+#define sxstate(x,y) socksstate(x,y)
 #endif
 
 
 /* always use this function to change state, to make debugging easier */
 static void socksstate(struct Curl_easy *data,
                        enum connect_t state
-#ifdef DEBUGBUILD
+#ifdef DEBUG_AND_VERBOSE
                        , int lineno
 #endif
 )
 {
   struct connectdata *conn = data->conn;
   enum connect_t oldstate = conn->cnnct.state;
-#if defined(DEBUGBUILD) && !defined(CURL_DISABLE_VERBOSE_STRINGS)
+#ifdef DEBUG_AND_VERBOSE
   /* synced with the state list in urldata.h */
   static const char * const statename[] = {
     "INIT",
     "SOCKS_INIT",
     "SOCKS_SEND",
@@ -144,11 +145,11 @@ static void socksstate(struct Curl_easy *data,
     /* don't bother when the new state is the same as the old state */
     return;
 
   conn->cnnct.state = state;
 
-#if defined(DEBUGBUILD) && !defined(CURL_DISABLE_VERBOSE_STRINGS)
+#ifdef DEBUG_AND_VERBOSE
   infof(data,
         "SXSTATE: %s => %s conn %p; line %d",
         statename[oldstate], statename[conn->cnnct.state], conn,
         lineno);
 #endif

Reply:
Done.